### PR TITLE
feat: add meta notification

### PR DESCRIPTION
**Pull request by yoshwata**
**Issue #, if available:**
It resembles to [1177](https://github.com/screwdriver-cd/screwdriver/issues/1177).

**Description of changes:**
I add guide about [meta notification feature](https://github.com/screwdriver-cd/notifications-slack/pull/19).

I confirm that this contribution is made under the terms of your choice and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: docs/user-guide/configuration/settings.md [hunk 1]
```diff
@@ -57,6 +57,8 @@ To enable Slack notifications to be sent as a result of build events, invite the
 
 You can also configure when to send a Slack notification, e.g. when the build status is `SUCCESS` and/or `FAILURE`. If no `statuses` field is set, the build will only send notifications for build failures. For a full list of statuses, see the [data-schema](https://github.com/screwdriver-cd/data-schema/blob/c2ea9b0372c6e62cb81e1f50602b751d0b10d547/models/build.js#L83-L96).
 
+To send data in steps as notification, [notification meta](https://docs.screwdriver.cd/user-guide/metadata.html#notification) is available.
+
```
Comment (wdstar) [inline docs/user-guide/configuration/settings.md:61]: `as a notification`

File: docs/user-guide/metadata.md [hunk 2]
```diff
@@ -190,3 +192,16 @@ jobs:
 
 Result:
 ![Label](./assets/label-meta.png)
+
+### Notification
```
Comment (wdstar) [inline docs/user-guide/metadata.md]: `Notifications` ?

File: docs/user-guide/configuration/settings.md (hunk 1 above)
Comment (yoshwata) [reply docs/user-guide/configuration/settings.md:61]: fixed~

File: docs/user-guide/metadata.md (hunk 2 above)
Comment (yoshwata) [reply docs/user-guide/metadata.md]: fixed~

File: docs/user-guide/metadata.md [hunk 3]
```diff
@@ -190,3 +192,16 @@ jobs:
...
+
+You can use [notification](./configuration/settings.html#slack) function from steps.
```
Comment (wahapo) [inline docs/user-guide/metadata.md]: Is this better?
`You can custom message for [notification](./configuration/settings.html#slack). Custom content is different for each notification plugins.`

Comment (yoshwata) [reply docs/user-guide/metadata.md]: Thanks, I fixed like you told.

**Review by wahapo — APPROVED** [review]

**Review by wdstar — APPROVED** [review]
LGTM

File: docs/user-guide/metadata.md [hunk 4]
```diff
@@ -24,6 +24,8 @@ toc:
       url: "#coverage-and-test-results"
     - title: <span class="menu-indent">Event Labels</span>
       url: "#event-labels"
+    - title: <span class="menu-indent">Notification</span>
```
Comment (tkyi) [inline docs/user-guide/metadata.md]:
```suggestion
    - title: <span class="menu-indent">Notifications</span>
```

File: docs/user-guide/metadata.md [hunk 5]
```diff
@@ -24,6 +24,8 @@ toc:
...
+      url: "#notification"
```
Comment (tkyi) [inline docs/user-guide/metadata.md]:
```suggestion
      url: "#notifications"
```

File: docs/user-guide/metadata.md [hunk 6]
```diff
@@ -190,3 +192,17 @@ jobs:
 
 Result:
 ![Label](./assets/label-meta.png)
+
+### Notifications
+
+You can customize [notification](./configuration/settings.html#slack) messages with meta.
+Meta keys are different for each notification plugins.
```
Comment (tkyi) [inline docs/user-guide/metadata.md]:
```suggestion
Meta keys are different for each notification plugin.
```

File: docs/user-guide/metadata.md [hunk 7]
```diff
@@ -190,3 +192,17 @@ jobs:
...
+
+Example screwdriver.yaml notifying with slack:
```
Comment (tkyi) [inline docs/user-guide/metadata.md]:
```suggestion
Example screwdriver.yaml notifying with Slack:
```

**Comment by tkyi** [comment]
A few minor comments
Also might be nice to put a screenshot of the new custom Slack notification

**Review by tkyi — APPROVED** [review]
👍 
image

**Pull request merged**